rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/RMServiceImpl.java

```diff
@@ -288,222 +258,146 @@ public String defineNodeSource(final String sessionId, final String nodeSourceNa
-            }
-        });
+    @Override
+    public String unlockNodes(String sessionId, Set<String> urls) throws RestServerException, ServiceException {
+        return executeFunction((restClient, strings) -> restClient.unlockNodes(sessionId, strings), urls, "unlock");
     }
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see
-     * org.ow2.proactive_grid_cloud_portal.rm.client.RMService#removeNodesource(java.lang.String,
-     * java.lang.String)
-     */
-    public String removeNodesource(final String sessionId, final String name, final boolean preempt)
+    @Override
+    public String removeNode(String sessionId, String url, boolean force) throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(restClient -> restClient.removeNode(sessionId, url, force));
+    }
+
+    @Override
+    public String removeNodesource(String sessionId, String name, boolean preempt)
             throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient) {
-                return restClient.removeNodesource(sessionId, name, preempt);
-            }
-        });
+        return executeFunctionReturnStreamAsString(restClient -> restClient.removeNodesource(sessionId, name, preempt));
     }
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see org.ow2.proactive_grid_cloud_portal.rm.client.RMService#releaseNode(java.lang.String,
-     * java.lang.String)
-     */
-    public String releaseNode(final String sessionId, final String url) throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient) {
-                return restClient.releaseNode(sessionId, url);
-            }
-        });
+    @Override
+    public String releaseNode(String sessionId, String url) throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(restClient -> restClient.releaseNode(sessionId, url));
     }
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see Service#getVersion()
-     */
+    @Override
     public String getVersion() throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient) {
-                return restClient.getVersion();
-            }
-        });
+        return executeFunctionReturnStreamAsString(RestClient::getVersion);
     }
 
     @Override
-    public String getMBeanInfo(final String sessionId, String name, final List<String> attrs)
+    public String getMBeanInfo(String sessionId, String name, List<String> attrs)
             throws RestServerException, ServiceException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)